src/Sarif/Converters/AndroidStudioConverter.cs

```diff
@@ -82,11 +101,13 @@ private void ProcessAndroidStudioLog(XmlReader xmlReader, IResultLogWriter outpu
                 var problem = AndroidStudioProblem.Parse(xmlReader, _strings);
                 if (problem != null)
                 {
-                    output.WriteResult(AndroidStudioConverter.ConvertProblemToSarifIssue(problem));
+                    results.Add(ConvertProblemToSarifIssue(problem));
```

Comment:
One thing about channeling results writing through WriteResult is that this is a useful way to glean ruleInfo and fileReference information. I.E., as results are written, the logger can remember what rules are encountered, what unique file references are seen and then automatically emit this data.